### PR TITLE
feat(StylelintConfig): Add shared stylelint config package WEB-990

### PR DESCRIPTION
**Pull request by codecaaron**
## Overview
Adds stylelint config package to lerna town.

This will also work on styled components.

### PR Checklist

- [ ] Related to Abstract designs:
- [x] Related to JIRA ticket: WEB-990
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### Description

- Adds `@codecademy/stylelint-config` npm package.
- Based off of https://github.com/codecademy-engineering/Codecademy/blob/master/.stylelintrc
- Adds 'stylelint-config-styled-components' to disable conflicting rules
- Adds recommended VSCode extension: `hex-ci.stylelint-plus` (the one distributed by stylelint didn't seem to work out of the box, however this does, happy to switch if people get the other one working easily).

Styled component config is basically just:
```js
module.exports = {
  "rules": {
    "value-no-vendor-prefix": true,
    "property-no-vendor-prefix": true,
    "no-empty-source": null,
    "no-missing-end-of-source-newline": null
  }
}
```

Further work:
- Full audit of the rules that we want. 
- Enable specific rulesets per package.
- Replace usage on Monolith with new package.
- Add lint verification scripts and tasks if we are happy with the rulesets.
- Deprecate SCSS lint?

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/stylelint-config@1.0.1-alpha.1627ec.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f20348626103b019c29c98a--gamut-preview.netlify.app>https://5f20348626103b019c29c98a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f20348626103b019c29c98a>Deploy Logs</a></p>

File: packages/stylelint-config/index.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    'block-closing-brace-newline-after': null,
+    'comment-empty-line-before': null,
+    'declaration-colon-newline-after': null,
+    'declaration-empty-line-before': null,
+    'font-family-no-missing-generic-family-keyword': null,
+    indentation: null,
+    'rule-empty-line-before': null,
+    'selector-combinator-space-before': null,
+    'selector-descendant-combinator-no-non-space': null,
+    'value-list-comma-newline-after': null,
```
Comment (JoshuaKGoldberg) [inline packages/stylelint-config/index.js:35]: Come to think of if, https://github.com/prettier/stylelint-prettier would let us get away without a bunch of these formatting rules. Can we get this in for the initial release so we don't have to breaking change it in later?

**Comment by codecaaron** [comment]
Closed in favor of https://github.com/Codecademy/client-modules/pull/921

**Pull request closed without merging**